plugins/analytics-module-segment/src/apis/implementations/AnalyticsApi/Segment.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Config } from '@backstage/config';
+import { AnalyticsApi, AnalyticsEvent } from '@backstage/core-plugin-api';
+import { AnalyticsBrowser } from '@segment/analytics-next';
+
+/**
+ * Segment provider for the Backstage Analytics API.
+ * @public
+ */
+export class SegmentAnalytics implements AnalyticsApi {
+  private readonly analytics: AnalyticsBrowser;
+  private readonly testMode: boolean;
+
+  /**
+   * Instantiate the implementation and initialize Segment client.
+   */
+  private constructor(options: { writeKey: string; testMode: boolean }) {
+    const { writeKey, testMode } = options;
+
+    this.testMode = !!testMode;
+    this.analytics = AnalyticsBrowser.load({ writeKey: writeKey });
+  }
+
+  /**
+   * Instantiate a fully configured Segment API implementation.
+   */
+  static fromConfig(config: Config) {
+    const writeKey = config.getString('app.analytics.segment.writeKey');
+    const testMode =
+      config.getOptionalBoolean('app.analytics.segment.testMode') ?? false;
+
+    return new SegmentAnalytics({
+      writeKey,
+      testMode,
+    });
+  }
+
+  async captureEvent(event: AnalyticsEvent) {
+    // Don't capture events in test mode.
+    if (this.testMode) {
+      return;
+    }
+
+    const { action, subject, value, context, attributes } = event;
+    const { routeRef, pluginId, extension, ...additionalContext } = context;
+
+    // Identify users.
+    if (action === 'identify') {
+      await this.analytics.identify(subject, {
+        email: attributes?.email,
+        name: attributes?.name,
+      });
+      return;
+    }
+
+    // Track page views.
+    if (action === 'navigate') {
+      await this.analytics.page();
+      return;
+    }
+
+    // Track other events.
+    await this.analytics.track('Event', {
```

Comment:
@alecjacobs5401 @knksmith57 -- Wanted to get your thoughts on how events are named going into Segment. This currently would use a fairly generic name `Event` -- but it may be worth trying to make this customizable (e.g. through a config value or custom handler) and/or more aligned with Segment's recommended [naming conventions](https://segment.com/academy/collecting-data/naming-conventions-for-clean-data/)

Reply:
Hey Andrew! Thanks for your patience, our whole team was at an offsite last week and we're all returning today. Let me sync up with them and the AJS team and get a good answer for this